codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/generators/PythonApplicationGenerator.kt

```diff
@@ -212,13 +209,22 @@ class PythonApplicationGenerator(
                 }
                 rustTemplate(
                     """
-                    let middleware_locals = #{pyo3_asyncio}::TaskLocals::new(event_loop);
-                    let service = #{tower}::ServiceBuilder::new()
-                        .boxed_clone()
-                        .layer(
-                            #{SmithyPython}::PyMiddlewareLayer::<#{Protocol}>::new(self.middlewares.clone(), middleware_locals),
-                        )
-                        .service(builder.build());
+                    let mut service = #{tower}::util::BoxCloneService::new(builder.build());
+
+                    {
+                        use #{tower}::Layer;
+                        tracing::debug!("adding middlewares to rust python router");
```

Comment:
I would put this (and the log statement below) at `trace` level rather than `debug`.
It's almost never going to be useful, so worth keeping it at the chattiest level.